examples/cert-manager/cert-manager.cue

```diff
@@ -0,0 +1,1185 @@
+            """
+			"for": string | *"1h"
+			labels: {
+				severity: "warning"
+			}
+			annotations: {
+				summary:     "The cert `{{ $labels.name }}` is {{ $value | humanizeDuration }} from expiry, it should have renewed over a week ago."
+				description: "The domain that this cert covers will be unavailable after {{ $value | humanizeDuration }}. Clients using endpoints that this cert protects will start to fail in {{ $value | humanizeDuration }}."
+				// TODO this is totally broken right now because it relies on a
+				// hardcoded uid for the particular dashboard. Polly provides the
+				// necessary namespacing information such that it should no longer be
+				// necessary to sling around uids like this - instead, this should be
+				// a reference to the namespaced name of the polly dashboard.
+				//
+				// That's the ideal, anyway - we'll have to see what we can actually
+				// accomplish :)
+				dashboard_url: header.params.grafanaExternalUrl + "/d/TvuRo2iMk/cert-manager"
+			}
+		}
+	}
+	CertManagerCertNotReady: {
+		group: "certificates"
+		alert: {
+			expr: """
+				max by (name, exported_namespace, namespace, condition) (
+				  certmanager_certificate_ready_status{condition!=\"True\"} == 1
+				)
+				"""
+			"for": string | *"10m"
+			labels: {
+				severity: "critical"
+			}
+			annotations: {
+				summary:       "The cert `{{ $labels.name }}` is not ready to serve traffic."
+				description:   "This certificate has not been ready to serve traffic for at least 10m. If the cert is being renewed or there is another valid cert, the ingress controller _may_ be able to serve that instead."
+				dashboard_url: header.params.grafanaExternalUrl + "/d/TvuRo2iMk/cert-manager"
+			}
+		}
+	}
+	CertManagerHittingRateLimits: {
+		group: "certificates"
+		alert: {
+			expr: """
+				sum by (host) (
+				  rate(certmanager_http_acme_client_request_count{status=\"429\"}[5m])
+				) > 0
+				"""
+			"for": string | *"5m"
+			labels: {
+				severity: 'critical'
+			}
+			annotations: {
+				summary:       "The cert `{{ $labels.name }}` is not ready to serve traffic."
+				description:   "This certificate has not been ready to serve traffic for at least 10m. If the cert is being renewed or there is another valid cert, the ingress controller _may_ be able to serve that instead."
+				dashboard_url: header.params.grafanaExternalUrl + "/d/TvuRo2iMk/cert-manager"
+			}
+		}
+	}
+	CertManagerAbsent: {
+		group: "cert-manager"
+		alert: {
+			expr:  "absent(up{job=\"\(header.params.certManagerJobLabel)s\"})"
+			"for": string | *"10m"
+			labels: {
+				severity: 'critical'
+			}
+			annotations: {
+				summary:       "Cert Manager has dissapeared from Prometheus service discovery."
+				description:   "New certificates will not be able to be minted, and existing ones can't be renewed until cert-manager is back."
+				dashboard_url: header.params.grafanaExternalUrl + "/d/TvuRo2iMk/cert-manager"
+			}
+		}
+	}
+}
+
+grafanaDashboards: v0: {
```

Comment:
Minor issue - would it make sense to structure code in a similar way we did it in mixins? 

put dashboards in separate files? With a pretty small pollypkg containing just 1 dashboard we already have a 1k+ LOC and it might get too long to be readable in case of larger pollypkgs/mixins.

Reply:
I think it would, yes.

we discussed this a bit on the last hack session - file layout relates heavily to how we namespace (URI) and generate code from a UI. The rules here can be simple, I think - handwaving, but basically:

- All pop files must be in the same directory
- All files must be in the same CUE package (unclear if we specify what that package must be)
- The emit value of the package must be an instance of `PollyPackage`
- Files created by UI-driven tools should follow some suffix or prefix pattern
- Files containing only some particular object type should follow some prefix or suffix pattern (we can lint this! 🎉)

Reply:
These rules have the effect of making package generation and linting pretty straightforward (standard CUE rules for loading a whole dir's package), removing any (I think?) unnecessary choice, render any within-the-pop importing unnecessary (unlike today's mixins). It should all be smooth and clear.

If you could post an issue about this, I'll use the next example/hack session as an opportunity to either do a new pop that conforms to these patterns, or make a new one 😄